RecoPixelVertexing/PixelTriplets/plugins/CellularAutomaton.cc

```diff
@@ -0,0 +1,112 @@
+
+#include "CellularAutomaton.h"
+
+template <unsigned int numberOfLayers>
+void CellularAutomaton<numberOfLayers>::create_and_connect_cells (std::vector<const HitDoublets*> doublets, const SeedingLayerSetsHits::SeedingLayerSet& fourLayers, const TrackingRegion& region, const float thetaCut, const float phiCut)
+{
+  unsigned int cellId = 0;
+  constexpr unsigned int numberOfLayerPairs =   numberOfLayers - 1;
+  float ptmin = region.ptMin();
+  float region_origin_x = region.origin().x();
+  float region_origin_y = region.origin().y();
+  float region_origin_radius = region.originRBound();
+  unsigned int layerPairId = 0;
+  auto innerLayerId = layerPairId;
+  auto outerLayerId = innerLayerId + 1;
+  auto numberOfDoublets = doublets[layerPairId]->size ();
+
+  isOuterHitOfCell[outerLayerId].resize(fourLayers[outerLayerId].hits().size());
+  theFoundCellsPerLayer[layerPairId].reserve (numberOfDoublets);
+  for (unsigned int i = 0; i < numberOfDoublets; ++i)
+  {
+    theFoundCellsPerLayer[layerPairId].emplace_back (doublets[layerPairId], i,  cellId,  doublets[layerPairId]->innerHitId(i), doublets[layerPairId]->outerHitId(i));
+
+    isOuterHitOfCell[outerLayerId][doublets[layerPairId]->outerHitId(i)].push_back (&(theFoundCellsPerLayer[layerPairId][i]));
```

Comment:
There is no reserve because we haven't still measured how many doublets start from the same hit wrt PU 35->70.
Do you suggest to reserve a reasonable size and then modify it later, or to leave like it is and then make a pull request with the final size?